### ISSUE TITLE
Design for metadata view on item show page

### ISSUE THREAD

**Issue by blalbrit** [sul-dlss/exhibits#799]
Per user feedback and team discussion, the endless scroll of full metadata on the item show page is a less-than-optimal user experience. Are there design options we could pursue for breaking sections out into tabs or collapsibles that might allow the user greater control over how much they see? @ggeisler - you may have thought about this in detail already, but we wanted to review possible approaches to this problem.

**Comment by mejackreed** [comment]
Also one thing came up is how should we display notes? The labeled notes display mentioned is a departure from how we display notes in SearchWorks and PURL and I just wanted to get some feedback from a design perspective.

https://github.com/sul-dlss/exhibits/pull/795#issuecomment-339495992

**Comment by camillevilla** [comment]
@ggeisler Here's what a manuscript show page looks like with a full set of labeled notes and bibliography. Things are getting pretty long:

![show-page](https://user-images.githubusercontent.com/5402927/32080052-57a45198-ba62-11e7-8526-5844508db4b1.gif)

## Added metadata
![cambridge__corpus_christi_college__ms_69__gregory_the_great__homiliae_xl_in_euangelia__-_test_-_stanford_university_libraries](https://user-images.githubusercontent.com/5402927/32080202-3b27401a-ba63-11e7-9bed-a5b4299db20f.png)


## Fixtures
- Manuscript: [cf386wt1778](purl.stanford.edu/cf386wt1778)
- [Collection of bibliography items](https://drive.google.com/file/d/0B30Uf0ba2atMVWZTek91SXRId2c/view?usp=sharing)

**Comment by camillevilla** [comment]
Per story time 11/2:
- @blalbrit will provide some additional MODS fixtures with examples of notes used outside of `<physicalDescription>`

**Comment by camillevilla** [comment]
Related tickets: #817, #571

**Comment by ggeisler** [comment]
Based on story time discussion on 11/9, here are initial mockups.

### Add button to trigger display of modal

The addition here is just adding a button to trigger the display of the modal with full metadata (note 2 in mockup). 

<img width="384" alt="modal-md-button" src="https://user-images.githubusercontent.com/101482/32629452-7bb13a7a-c556-11e7-8222-d03fe7f38cc2.png">

### Create modal to display full metadata of the item
The idea for the modal is to present all metadata in the item's MODS record.

Details of the modal content and layout pending discussion with and potential work from @jvine to ensure the enhancements necessary to the MODS Display gem to support this modal are compatible with the use of that gem in other SUL contexts.

Initial display, with the related items (`<relatedItem type="constituent">`) collapsed:

<img width="381" alt="modal-md-collapsed" src="https://user-images.githubusercontent.com/101482/32629655-2a4264a6-c557-11e7-89e3-1bf6e35768e0.png">

----

Example where the user has expanded the first related item to see its nested fields:

<img width="373" alt="modal-md-expanded" src="https://user-images.githubusercontent.com/101482/32629677-3c778e76-c557-11e7-94ab-f8ddb7d05559.png">

**Comment by ggeisler** [comment]
@jvine In case it helps, https://purl.stanford.edu/hp566jq8781.mods is an item with multiple related items.

**Comment by jvine** [comment]
Still working on grouping (or not) of metadata, but have some details for discussion/possible work:

1. "Contents" in the metadata popup should also be in list format as on the show page (i.e. -- converted to `<li>`)  but without show/hide functionality. (Differs from Gary's mockup.)

> **Before**
> ![Screen Shot 2017-11-15 at 4.25.02 PM.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/55e7873278db271d00ed1647/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c57e0b76226455f66b7191a37cf1102e4f5f5d5ecc3c4a4356bcbbd671f3e57862804e0eead6685622923511afbc821db600b3208b498ce4b5db4cd2f42f67b6d5db1063474f4dcbed5727a29941c95df74be64454ad1b3362e6732431.png)
---
> **After**
> ![Screen Shot 2017-11-15 at 4.25.59 PM.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/55e7873278db271d00ed1647/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c57e0b76226455f66b7191a37cf1102e4f5a565ecc3c4a4356bcbbd671f3e57862804e0eead6685622923511afbc821db600b3208b498ce4b5db4cd2f42f67b6d5db1063474f4dcbed5727a19e41c95df74be64457a4133e67e1752431.png)
---

2. Constituent items are currently (in mods_display) displaying `<titleinfo>` and `<titleInfo ... type="alternative">`as a combined string. The primary `<titleinfo>` should be displayed alone, and the remaining metadata (`<titleInfo ... type="alternative">` and everything else) should be displayed with a show/hide toggle (as in Gary's mockup).

3. The title for "Related items" should be conditionally based on the type of related item (constituent, host, etc.). 

4. Ideally, the entire "Related items" section will have an "expand all" option - since the number of related items can be long, and if that's what the researcher is there to see, expanding each individually will be tedious. 

5. My interpretation of Gary's mockup is that the popup is a fixed height within the viewport, while the `modal-body` div is scrollable. Close and Download buttons are intended to be always visible / fixed at the bottom.

**Comment by drh-stanford** [comment]
@jvine I'm a bit confused by the placement and labeling of the "Related items". There's constituent, host, etc. types for the related items in the MODS. For (3) and (4) should there be different sections for each type? Also, as I understand it, all related items will go in the Related items section, not in the core block (which is does now).

**Comment by jvine** [comment]
@drh the specific ordering/grouping/labeling of the metadata is still WIP. I just wanted to pull out that:

3. Currently, host and constituent related items are bundled together with one generic label. They will need separate labels to group them by type, whether they are listed in the core block or in a separate section. 

4. The mockups show a show/hide interaction for the individual items that have nested metadata. Added feature: if there's more than one related item (in a group of the same type), there should be an "expand all" option for the group.

Should some/all of these items be separate tickets? Should I wait until I've finished new mockups?
